packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -172,11 +291,23 @@ export default class CodeGen {
             textConcatenation = t.binaryExpression('+', textConcatenation, mappedValues[i]);
         }
 
-        return this._renderApiCall(RENDER_APIS.text, [textConcatenation]);
+        const apiCall = this._renderApiCall(RENDER_APIS.text, [textConcatenation]);
+
+        return isHoisted ? this.genHoistedNode(apiCall) : apiCall;
+    }
+
+    genHoistedNode(expr: t.Expression): t.CallExpression {
+        this.hoistedNodes.push(expr);
+
+        return this._renderApiCall(RENDER_APIS.setOwner, [
+            t.identifier(`$hoisted${this.hoistedNodes.length}`),
```

Comment:
In my hoisting, I used the prefix `stc_`. It would be nice to use consistent naming (although not a big deal).

Reply:
Well, this `hoisted` is different from `stc_`. It may be the case that you have both `hoisted$` and `stc_`. `hoisted` stores vnodes, while `stc_` store static expressions. I like that the names are different.